packages/tasit-act/src/TasitAct.test.js

```diff
@@ -0,0 +1,137 @@
+import { Contract } from "./TasitAct";
+import Account from "tasit-account";
+import { expect, assert } from "chai";
+import { waitForEvent, mineBlocks } from "./helpers/utils.js";
+
+// Note: SimpleStorage.json is originally genarated by `tasit-contracts` and was pasted here manually
+// See https://github.com/tasitlabs/TasitSDK/issues/45
+import { abi as contractABI } from "./helpers/SimpleStorage.json";
+
+// Note: Under the current `tasit-contracts` setup SimpleStorage aways will deployed with this address
+// See https://github.com/tasitlabs/TasitSDK/pull/59#discussion_r242258739
+const contractAddress = "0x6C4A015797DDDd87866451914eCe1e8b19261931";
+
+describe("Contract", function() {
+  let simpleStorage, wallet;
+
+  beforeEach("should connect to an existing contract", async () => {
+    // Account creates a wallet, should it create an account object that encapsulate the wallet?
+    // TasitAcount.create()
+    // > Acount { wallet: ..., metaTxInfos..., etc }
+    wallet = Account.createFromPrivateKey(
+      "0x11d943d7649fbdeb146dc57bd9cfc80b086bfab2330c7b25651dbaf382392f60"
+    );
+
+    simpleStorage = new Contract(contractAddress, contractABI);
+    expect(simpleStorage).to.exist;
+    expect(simpleStorage.address).to.equal(contractAddress);
+    expect(simpleStorage.getValue).to.exist;
+    expect(simpleStorage.setValue).to.exist;
+    // Events are not implemented yet
+    //expect(simpleStorage.ValueChanged).to.exist;
+  });
+
+  it("should throw error when instantiated with invalid args", async () => {
+    try {
+      new Contract();
+      assert(false, "constructor without address and ABI");
+
+      new Contract(contractAddress);
+      assert(false, "constructor without ABI");
+
+      new Contract("invalid address");
+      assert(false, "constructor with invalid address");
+
+      new Contract("invalid address", contractABI);
+      assert(false, "constructor with invalid address and valid ABI");
+
+      new Contract(contractAddress, "invalid abi");
+      assert(false, "constructor with valid address and invalid ABI");
+    } catch (e) {
```

Comment:
I've done that way because each of these requires a try/catch block, so inside of one it the code seems more clear. On the other hand as separated test cases each should live in isolated `it` block. I'll do that.